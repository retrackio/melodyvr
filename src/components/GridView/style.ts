import styled from 'styled-components';

export const Container = styled.div`
  max-width: 100%;
  margin: 0 auto;
`;

export const Row = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
`;

export const Item = styled.div`
  flex-basis: 25%;
  box-sizing: border-box;
  /* position: relative; */
  padding: 10px;

  @media (max-width: 1024px) {
    flex-basis: 33.3%;
  }

  @media (max-width: 768px) {
    flex-basis: 50%;
  }

  @media (max-width: 600px) {
    flex-basis: 100%;
  }
`;
