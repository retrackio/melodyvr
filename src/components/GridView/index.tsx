import React from 'react';
import * as Styled from './style';

interface Props {
  items: React.ReactNode[];
}

const GridView = ({ items }: Props) => (
  <Styled.Container>
    <Styled.Row>
      {items.map((item, index) => (
        <Styled.Item key={`$grid_item_${index}`}>{item}</Styled.Item>
      ))}
    </Styled.Row>
  </Styled.Container>
);

export default GridView;
