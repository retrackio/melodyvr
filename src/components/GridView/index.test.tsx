import React from 'react';
import { shallow } from 'enzyme';
import GridView from './';

const mockedItems: React.ReactNode[] = [
  <div key="1">Item 1</div>,
  <div key="2">Item 2</div>,
];

describe('<GridView />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GridView items={mockedItems} />);
    expect(wrapper).toMatchSnapshot();
  });
});
