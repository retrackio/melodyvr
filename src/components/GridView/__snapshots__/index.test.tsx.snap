// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridView /> renders correctly 1`] = `
<style__Container>
  <style__Row>
    <style__Item
      key="$grid_item_0"
    >
      <div
        key="1"
      >
        Item 1
      </div>
    </style__Item>
    <style__Item
      key="$grid_item_1"
    >
      <div
        key="2"
      >
        Item 2
      </div>
    </style__Item>
  </style__Row>
</style__Container>
`;
